pkgs/development/compilers/emscripten/ports.nix

```diff
@@ -0,0 +1,136 @@
+    emscriptenCachePackages = [ name ] ++ map (extra: "${name}${extra}") extraSuffixes;
+  };
+
+portsSpecs = {
+  # URL and extras must match those that embuilder.py expects
+  sdl2 = {
+    url = "https://github.com/libsdl-org/SDL/archive/release-2.24.0.zip";
+    sha256 = "sha256-CtOLNwUQa2fLt+CN0lcBimwAMAhveqldLvzsl2IgR9g=";
+    extraSuffixes = [ "-mt" ];
+  };
+  boost_headers = {
+    url = "https://github.com/emscripten-ports/boost/releases/download/boost-1.75.0/boost-headers-1.75.0.zip";
+    sha256 = "sha256-ezwSMlIzchKAcxy+ixFxP0clEWngaT5waMSSp082z84=";
+  };
+  sdl2_gfx = {
+    url = "https://github.com/libsdl-org/SDL/archive/release-2.24.0.zip";
+    sha256 = "sha256-CtOLNwUQa2fLt+CN0lcBimwAMAhveqldLvzsl2IgR9g=";
+  };
+  sdl2_image  = {
+    url = "https://github.com/libsdl-org/SDL/archive/release-2.24.0.zip";
+    sha256 = "sha256-CtOLNwUQa2fLt+CN0lcBimwAMAhveqldLvzsl2IgR9g=";
+  };
+  sdl2_mixer  = {
+    url = "https://github.com/libsdl-org/SDL/archive/release-2.24.0.zip";
+    sha256 = "sha256-CtOLNwUQa2fLt+CN0lcBimwAMAhveqldLvzsl2IgR9g=";
+  };
+  sdl2_net  = {
+    url = "https://github.com/libsdl-org/SDL/archive/release-2.24.0.zip";
+    sha256 = "sha256-CtOLNwUQa2fLt+CN0lcBimwAMAhveqldLvzsl2IgR9g=";
+  };
+  sdl2_ttf  = {
+    url = "https://github.com/libsdl-org/SDL/archive/release-2.24.0.zip";
+    sha256 = "sha256-CtOLNwUQa2fLt+CN0lcBimwAMAhveqldLvzsl2IgR9g=";
+  };
+  bullet  = {
+    url = "https://github.com/emscripten-ports/bullet/archive/version_1.zip";
+    sha256 = "sha256-pcmCUE++CXY+xegV8prSrWq79iwDnKmVEhWHIVaeH7Q=";
+  };
+  bzip2  = {
+    url = "https://github.com/emscripten-ports/bzip2/archive/1.0.6.zip";
+    sha256 = "sha256-LKYisuMW9OGMJX04epqTYQ9RkUdLTRXZ7nGF1J8V7oY=";
+  };
+  cocos2d = {
+    url = "https://github.com/libsdl-org/SDL/archive/release-2.24.0.zip";
+    sha256 = "sha256-CtOLNwUQa2fLt+CN0lcBimwAMAhveqldLvzsl2IgR9g=";
+  };
+  giflib  = {
+    url = "https://storage.googleapis.com/webassembly/emscripten-ports/giflib-5.2.1.tar.gz";
+    sha256 = "sha256-814dyiemFtSJktrlTZ8YE2/wFvkNjWPJERcYlDPkpNo=";
+  };
+  harfbuzz  = {
+    url = "https://github.com/emscripten-ports/FreeType/archive/version_1.zip";
+    sha256 = "sha256-mawoDIWaYHHw+8bFgAhSQkx5DEooVxW1yZhGcxyygQM=";
+    extraSuffixes = [ "-mt" ];
+  };
+  icu  = {
+    url = "https://github.com/unicode-org/icu/releases/download/release-68-2/icu4c-68_2-src.zip";
+    sha256 = "sha256-4SDmJ4ISORm6K31OTtD9ik8QQuOte9i2jYaWjluQWnI=";
+    extraSuffixes = [ "-mt" ];
+  };
+  libjpeg  = {
+    url = "https://storage.googleapis.com/webassembly/emscripten-ports/jpegsrc.v9c.tar.gz";
+    sha256 = "sha256-SZ0g9UsM8VcXu2DfF/k9FUa6V7W2f5GPGQk4buWdU4s=";
+  };
+  libmodplug  = {
+    url = "https://github.com/jancc/libmodplug/archive/v11022021.zip";
+    sha256 = "sha256-hzXN30IocklLty8daZSF5rHQJ4xHKLWzCQP7o0m5Z0s=";
+  };
+  libpng  = {
+    url = "https://github.com/libsdl-org/SDL/archive/release-2.24.0.zip";
```

Comment:
At least this URL is duplicated. Not sure if it is worth to dedupe. Also can't we re-use the src's from our packages?

Reply:
I've fixed the duplicate URLs. The src's from nixpkgs cannot be used here, the `url` in this file point to the exact sources that `embuilder.py` requires.